omc3/nxcals/knob_extraction.py

```diff
@@ -0,0 +1,274 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from datetime import datetime, timedelta, timezone
+
+import jpype
+import pandas as pd
+from pyspark.sql import SparkSession
+from pyspark.sql import functions as f
+from pyspark.sql.window import Window
```

Comment:
pyspark is not in the requirements. And I think `SparkSession` should be in TYPE_HINTING